system/greenhouse-extras-storage/templates/metrics-ingress-grpc.yaml

```diff
@@ -0,0 +1,39 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  annotations:
+    disco: "true"
+    kubernetes.io/tls-acme: "true"
+
+    ingress.kubernetes.io/auth-tls-secret: {{ required ".Values.prometheus.auth.tlsSecret missing" .Values.prometheus.auth.tlsSecret | quote }}
+    nginx.ingress.kubernetes.io/auth-tls-secret: {{ required ".Values.prometheus.auth.tlsSecret missing" .Values.prometheus.auth.tlsSecret | quote }}
+    ingress.kubernetes.io/auth-tls-verify-depth: {{ required ".Values.prometheus.auth.tlsVerifyDepth missing" .Values.prometheus.auth.tlsVerifyDepth | quote }}
+    nginx.ingress.kubernetes.io/auth-tls-verify-depth: {{ required ".Values.prometheus.auth.tlsVerifyDepth missing" .Values.prometheus.auth.tlsVerifyDepth | quote }}
+    ingress.kubernetes.io/auth-tls-verify-client: {{ required ".Values.prometheus.auth.tlsVerifyClient missing" .Values.prometheus.auth.tlsVerifyClient | quote }}
+    nginx.ingress.kubernetes.io/auth-tls-verify-client: {{ required ".Values.prometheus.auth.tlsVerifyClient missing" .Values.prometheus.auth.tlsVerifyClient | quote }}
+
+    ingress.kubernetes.io/backend-protocol: GRPC
+    ingress.kubernetes.io/service-upstream: "true"
+    ingress.kubernetes.io/ssl-redirect: "true"
+    nginx.ingress.kubernetes.io/backend-protocol: GRPC
+    nginx.ingress.kubernetes.io/service-upstream: "true"
+    nginx.ingress.kubernetes.io/ssl-redirect: "true"
+  name: metrics-grpc
+  namespace: {{ required ".Values.thanos.namespace is unset" .Values.thanos.namespace }}
+spec:
+  ingressClassName: nginx
+  rules:
+  - host: metrics-grpc.st1.{{ .Values.global.region }}.cloud.sap
+    http:
+      paths:
+      - backend:
+          service:
+            name: thanos-{{ .Values.global.cluster }}-query
+            port:
+              name: grpc
+        path: /
+        pathType: Prefix 
+  tls:
+  - hosts:
+    - metrics-grpc.st1.{{ .Values.global.region }}.cloud.sap
+    secretName: tls-metrics-grpc.st1.{{ .Values.global.region }}.cloud.sap
```

Comment:
Can we provide those grpc Ingresses also with the Thanos plugin?

Reply:
Talked about this offline, currently not.